### PR TITLE
FreeBSD, OpenBSD thread renaming.

### PR DESCRIPTION
**Pull request by philmb3487**
Add support for thread renaming on FreeBSD, OpenBSD.

### CODE REVIEW

**Comment by sinetek** [comment]
Also possible would be NetBSD and DragonFlyBSD. I still have to test those, but I know the functions exist, just a matter of adding the right preprocessor macro's.

**Comment by laanwj** [comment]
utACK

File: src/util.cpp [hunk 1]
```diff
@@ -7,6 +7,11 @@
 #include "config/bitcoin-config.h"
 #endif
 
+#if (defined(__FreeBSD__) || defined(__OpenBSD__))
```
Comment (Diapolo) [inline src/util.cpp]: IMHO this should be moved below ``#include <stdarg.h>`` and also take into account what we have there Win/Linux def/ifdef etc. (I didn't check that).

Comment (sinetek) [inline src/util.cpp]: I suppose you're right, however there is namespace pollution in the bitcoin includes, so if I move it too much down stuff breaks dramatically in the openbsd sys/* includes.

**Comment by jgarzik** [comment]
ut ACK

**Comment by sipa** [comment]
Concept ACK

**Comment by theuni** [comment]
Generally, I'd rather see this tested in configure and used via
```
#if defined(HAS_PTHREAD_SET_NAME_NP)
```
These BSD defines tend to become messy otherwise. Plus, we'd automatically get support for dragonflybsd/netbsd/etc. As a rule, test for the feature, not the host.

I'm not sure it's worth the complication for this case though, because there are so many incompatible versions of the function.

ut ACK, new macros can be added if it proves necessary.

**Comment by sinetek** [comment]
note that this is dangerous though, because one BSD in particular likes to randomly reshuffle arguments around (I'll name it, the NetBSD saga with OpenSSL) and the others have subtle differences. In this case we'll need many conditionals anyway, since there's about 8 functions across all os's.

**Pull request merged**